doc/rules/function_notation/phpdoc_to_param_type.rst

```diff
@@ -30,6 +30,16 @@ Allowed types: ``bool``
 
 Default value: ``true``
 
+``union_types``
+~~~~~~~~~~~~~~~
+
+Fix also union types; may have unexpected behaviour due to PHP bad type coercion
```

Comment:
What do you mean by "unexpected behaviour", shouldn't it be fixed instead?

Reply:
@mvorisek 

I have copied the description from https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.46.0/src/AbstractPhpdocToTypeDeclarationFixer.php#L74.

Reply:
The comment probably meant `float` accepting `int` even with strict types. Not sure what other types, would be good to specify that in the `scalar_types` description explicitly.